src/bin/cql-stress-cassandra-stress/settings/command/user.rs

```diff
@@ -147,32 +152,43 @@ impl UserParams {
             keyspace_definition,
             table,
             table_definition,
-            queries,
+            mut queries,
         } = handles.profile.get().unwrap();
+        let queries_ratio = handles.ratio.get().unwrap();
+        let clustering = handles.clustering.get().unwrap();
 
-        let queries = queries
+        let queries_payload = queries_ratio
             .into_iter()
-            .map(|(query_name, query_def)| {
-                let query_def = query_def.into_query_definition();
-                match query_def {
-                    Ok(query_def) => Ok((query_name, query_def)),
-                    Err(e) => Err(e.context("Failed to parse query definition")),
-                }
-            })
+            .map(
+                |(query_name, weight)| -> Result<(String, (QueryDefinition, f64))> {
+                    let query_def = queries
+                        .remove(&query_name)
+                        .ok_or_else(|| {
+                            anyhow::anyhow!("Unrecognized query name in ratio map: {}", query_name)
+                        })?
+                        .into_query_definition()
+                        .context("Failed to parse query definition")?;
+
+                    Ok((query_name, (query_def, weight)))
+                },
+            )
```

Comment:
Is it an intended semantics change that now queries present in `queries` but without weights specified in `ratio` are ignored (i.e. not put in the returned map)?

Reply:
Yes, such operations won't be sampled anyway so there is no point in holding info about them. This just means that the user defined an operation in the yaml file, but decided not to use it during the runtime. The use case for this might be to have a single yaml file for multiple stress scenarios (e.g. each scenario executes only one operation from the yaml file).

For a moment, I thought that maybe we should fail parsing if there is an operation present in `queries` but not in `queries_ratio`. However, I think this could be annoying for users who want to define a single yaml file and reuse it for multiple stress runs.

Reply:
+ This is the behaviour of original c-s.